internal/literals/literals.go

```diff
@@ -97,74 +94,60 @@ func Obfuscate(file *ast.File, info *types.Info, fset *token.FileSet, ignoreObj
 				return true
 			}
 
-			data := make([]byte, 0, len(x.Elts))
+			data := make([]byte, 0, len(node.Elts))
 
-			for _, el := range x.Elts {
-				lit, ok := el.(*ast.BasicLit)
-				if !ok {
+			for _, el := range node.Elts {
+				elType := info.Types[el]
+
+				if elType.Value == nil || elType.Value.Kind() != constant.Int {
 					return true
 				}
-				var value byte
-				if lit.Kind == token.CHAR {
-					val, err := strconv.Unquote(lit.Value)
-					if err != nil {
-						panic(fmt.Sprintf("cannot unquote character: %v", err))
-					}
-
-					value = byte(val[0])
-				} else {
-					val, err := strconv.ParseUint(lit.Value, 0, 8)
-					if err != nil {
-						panic(fmt.Sprintf("cannot parse integer: %v", err))
-					}
-
-					value = byte(val)
+
+				value, ok := constant.Uint64Val(elType.Value)
+				if !ok {
+					panic(fmt.Sprintf("cannot parse byte value: %v", elType.Value))
 				}
 
-				data = append(data, value)
+				data = append(data, byte(value))
 			}
 
 			if arrayLen > 0 {
-				cursor.Replace(withPos(obfuscateByteArray(data, arrayLen), x.Pos()))
+				cursor.Replace(withPos(obfuscateByteArray(data, arrayLen), node.Pos()))
 			} else {
-				cursor.Replace(withPos(obfuscateByteSlice(data), x.Pos()))
+				cursor.Replace(withPos(obfuscateByteSlice(data), node.Pos()))
 			}
+		}
 
-			return true
-
-		case *ast.BasicLit:
-			switch cursor.Name() {
-			case "Values", "Rhs", "Value", "Args", "X", "Y", "Results", "Elts":
-			default:
-				return true // we don't want to obfuscate imports etc.
-			}
+		return true
+	}
 
-			if x.Kind != token.STRING {
-				return true
-			}
-			if len(x.Value) > maxSizeBytes {
-				return true
-			}
-			typeInfo := info.TypeOf(x)
-			if typeInfo != types.Typ[types.String] && typeInfo != types.Typ[types.UntypedString] {
-				return true
-			}
-			value, err := strconv.Unquote(x.Value)
-			if err != nil {
-				panic(fmt.Sprintf("cannot unquote string: %v", err))
-			}
+	prevUsedImports := make(map[string]bool)
+	for _, imp := range file.Imports {
+		path, err := strconv.Unquote(imp.Path.Value)
+		if err != nil {
+			panic(err)
+		}
+		prevUsedImports[path] = astutil.UsesImport(file, path)
```

Comment:
When would an existing import ever be unused? Presumably that would be a typechecking error, so we wouldn't get to this stage at all. If there's something I'm missing, perhaps document why this is needed, because it's not obvious to me at least.

Reply:
Repo: github.com/user/somepackage

somepackage.go:
```go
package anotherpackage
```


main.go:
```go

package main

import "github.com/user/somepackage"

func main(){
    anotherpackage.Foo()
}
```

This can't be detected via ast and is therefore "unused".

Reply:
Ah, yikes. Seems like this warrants a test case then, as well as inline docs.

Reply:
This is already tested in [imports.txt](https://github.com/burrowers/garble/blob/b5bef981ee28e4f09f5c655247cb3ba921798443/testdata/scripts/imports.txt#L154) (was failing before), will add docs